hardware_interface/include/hardware_interface/actuator_interface.hpp

```diff
@@ -72,6 +72,49 @@ class ActuatorInterface
   virtual
   std::vector<CommandInterface> export_command_interfaces() = 0;
 
+  /// Prepare for a new command interface switch.
+  /**
+   * Prepare for any mode-switching required by the new command interface combination.
+   *
+   * \note this is a non-realtime evaluation of whether a set of command interface claims are
+   * possible, and call to start preparing data structures for the upcoming switch that will occur.
+   * \note switching relevant for other hardware interfaces is also evaluated by this function,
+   * so if an interface key is not relevant for this hardware the function should return
+   * return_type::OK.
+   * \param[in] start_interfaces vector of string identifiers for the command interfaces starting.
+   * \param[in] stop_interfaces vector of string identifiers for the command interfacs stopping.
+   * \return return_type::OK if the new command interface combination can be prepared,
+   * or if the interface key is not relevant to this system. Returns return_type::ERROR otherwise.
+   */
+  virtual
+  return_type prepare_command_mode_switch(
+    const std::vector<std::string> & /*start_interfaces*/,
+    const std::vector<std::string> & /*stop_interfaces*/)
+  {
+    return return_type::OK;
+  }
+
+  // Perform switching to the new command interface.
+  /**
+   * Perform the mode-switching for the new command interface combination.
+   *
+   * \note this is part of the realtime update loop, and should be fast.
+   * \note switching relevant for other hardware interfaces is also evaluated by this function,
+   * so if an interface key is not relevant for this hardware the function should return
+   * return_type::OK.
+   * \param[in] start_interfaces vector of string identifiers for the command interfaces starting.
+   * \param[in] stop_interfaces vector of string identifiers for the command interfacs stopping.
+   * \return return_type::OK if the new command interface combination can be switched to,
```

Comment:
Is there possible to have a mix of relevant and irrelevant interfaces?

Reply:
Yup, see previous comment, I'll try to reword this so that it becomes clearer.